e2e-tests/tests/nextjs.test.ts

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable jest/expect-expect */
+import { Integration } from '../../lib/Constants';
+import {
+  checkEnvBuildPlugin,
+  cleanupGit,
+  KEYS,
+  revertLocalChanges,
+} from '../utils';
+import { startWizardInstance } from '../utils';
+import {
+  checkFileContents,
+  checkFileExists,
+  checkIfBuilds,
+  checkIfRunsOnDevMode,
+  checkIfRunsOnProdMode,
+  checkPackageJson,
+} from '../utils';
+import * as path from 'path';
+
+describe('NextJS', () => {
+  const integration = Integration.nextjs;
+  const projectDir = path.resolve(
+    __dirname,
+    '../test-applications/nextjs-test-app',
+  );
+
+  beforeAll(async () => {
+    const wizardInstance = startWizardInstance(integration, projectDir);
+    const packageManagerPrompted = await wizardInstance.waitForOutput(
+      'Please select your package manager.',
+    );
+
+    if (packageManagerPrompted) {
+      // Selecting `yarn` as the package manager
+      wizardInstance.sendStdin(KEYS.DOWN);
+      wizardInstance.sendStdin(KEYS.ENTER);
+    }
+
+    const routeThroughNextJsPrompted = await wizardInstance.waitForOutput(
+      'Do you want to route Sentry requests in the browser through your Next.js server',
+      {
+        timeout: 240_000,
+      },
+    );
```

Comment:
Is this order of operations prone to race conditions? Isn't it possible that the wizard emits "Do you want to ..." before we were able to attach the .waitForOutput() call.

I know it's a pain in the butt and makes the tests super unreadable, but we should probably err on the side of caution and attach the .waitForOutput before we send the `ENTER` key, and then afterwards await it. Does that make sense? We have the same pattern in the JS repo E2E tests quite often.

Reply:
Updated 👍 https://github.com/getsentry/sentry-wizard/pull/684/commits/4a6c56aa261943e850e62ce297e5b81ebe85a5ac